ion-schema/src/violation.rs

```diff
@@ -61,13 +61,11 @@ impl Violation {
     pub fn flattened_violations(&self) -> Vec<&Violation> {
         let mut flattened_violations = Vec::new();
         self.flatten_violations(&mut flattened_violations);
+        flattened_violations.sort_by(|a, b| (a.constraint).cmp(&b.constraint));
```

Comment:
I think the fields used for sorting needs to be consistent with the fields used for equality, or else we could end up with weird bugs.

Reply:
So equality uses flattened_violations itself. If we use it for sorting as well it would end up in infinite recursive calls.

Reply:
Sure, but I think we could get away with not sorting by flattened violations because if you flatten it once, all of the returned violations have no children (and flattening a childless violation is equivalent to an identity function).

All of the other fields need to be included in the sort.

Reply:
Returning a `HashSet` instead.